linchpin/cli/__init__.py

```diff
@@ -119,6 +119,43 @@ def lp_init(self, providers=['libvirt']):
             self.ctx.log_state('Error: {0}'.format(e))
             sys.exit(1)
 
+    def _write_to_inventory(self, inv_path=None, inv_format="cfg"):
+        latest_run_data = self._get_run_data_by_txid()
+        try:
+            for t_id in latest_run_data:
+                targets = latest_run_data[t_id]["targets"][0]
+                # if there are multiple targets mentioned in pinfile
+                # the multiple inventory files are being generated
+                inv_file_count = 0 if len(targets) > 1 else False
+                for name in targets:
+                    lt_data = targets[name]["inputs"]["layout_data"]
+                    layout = lt_data["inventory_layout"]
+                    inventory_path = targets[name]["inventory_path"]
+                    if inv_path and inv_file_count is not False:
+                        inventory_path = inv_path + str(inv_file_count)
+                    topology_outputs = targets[name]["topology_outputs"]
+                    inventory = self.generate_inventory(topology_outputs,
+                                                        layout,
+                                                        inv_format=inv_format)
+                    # if inv_path is explicitly mentioned it is used
+                    if inv_path:
+                        inventory_path = inv_path
+                    # if there are multiple targets based on number of targets
+                    # multiple files are generated with suffixes
+                    if inv_path and isinstance(inv_file_count, int):
+                        inventory_path = inv_path + "." + str(inv_file_count)
+                        with open(inventory_path, 'w') as the_file:
+                            the_file.write(inventory)
+                        inv_file_count += 1
+                    else:
+                        with open(inventory_path, 'w') as the_file:
+                            the_file.write(inventory)
+        except Exception as e:
+            self.ctx.log_state('Error: {0}'.format(e))
+            sys.exit(1)
+        return True
+
```

Comment:
Will this function eventually write to STDOUT to take advantage of the dynamic inventory option that ansible can use?

Reply:
we could always make it do that.

Reply:
And I guess in the STDOUT case, it might be better to put that with the [_handle_results function](https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/linchpin/shell/__init__.py#L24) as we discussed below.